src/networking.c

```diff
@@ -3728,17 +3821,159 @@ int canParseCommand(client *c) {
     return 1;
 }
 
+/* Pops a commands from the command queue and sets it as the client's current
+ * command. Returns true on success and false if the queue was empty. */
+static bool consumeCommandQueue(client *c) {
+    cmdQueue *queue = &c->cmd_queue;
+    if (queue->off >= queue->len) return false;
+    parsedCommand *p = &queue->cmds[queue->off++];
+    c->read_flags |= p->read_flags;
```

Comment:
```suggestion
    c->read_flags = p->read_flags;
```
Was this intentional? I think this looks safe, but it might subtly introduce bugs in the future if some flag is preserved .

Reply:
Maybe just documenting it would be enough. It's confusing that some of these are command flags (READ_FLAGS_PARSING_COMPLETED ) and some of these are client state flags (AUTH_REQUIRED)